src/allmydata/scripts/create_node.py

```diff
@@ -221,7 +225,9 @@ def write_node_config(c, config):
         c.write("\n")
 
     c.write("[node]\n")
-    nickname = argv_to_unicode(config.get("nickname") or "")
+    nickname = config.get("nickname") or u""
+    if not isinstance(nickname, unicode):
```

Comment:
When is it not?  Is it only for the `u""` default above?  If so, I suggest `b""` as a default and drop the conditional (ie, unconditionally decode).